tensorflow_datasets/scripts/replace_fake_images.py

```diff
@@ -114,10 +112,10 @@ def rewrite_tar(root_dir, tar_filepath):
       tar.extractall(path=temp_dir)
 
     rewrite_dir(temp_dir)  # Recursivelly compress the archive content
-
-    # Converting into tarfile again to decrease the space taken by the file-
-    with tarfile.open(tar_filepath, 'w' + extension) as tar:
-      tar.add(temp_dir, recursive=True)
+    
+    os.remove(tar_filepath) # Remove original tar file as dupicates with the extension tar.gz will be created
+    
+    shutil.make_archive(tar_filepath[:-4], 'gztar', temp_dir) # Copies the directory structure of the extracted tar file and stores it back gztar
```

Comment:
`[:-4]` won't work with `.tar.gz`, `.tgz` files.
Also, we should keep the same compression than the original archive. For instance, `.tgz` should use `gz` compression, but I'm not sure `.tar` should